src/main/webapp/ts/profileBuilder.ts

```diff
@@ -1,39 +1,59 @@
-type authInfoFromServlet = {
+type authInfo = {
     loginUrl: string;
     logoutUrl: string;
     hasProfile: boolean;
 };
 
 function setForm(value: string) {
-    const menteeForm =  document.getElementById('Mentee');
+    const menteeForm = document.getElementById('Mentee');
     const mentorForm = document.getElementById('Mentor');
     const userTypeButton = document.getElementById('user-input');
 
-    if(menteeForm && mentorForm && userTypeButton){
-        if(value == 'Mentee') {
-            menteeForm.setAttribute('style','display:block;');
-            mentorForm.setAttribute('style','display:none;');
-            userTypeButton.setAttribute('value','Mentee');
+    const menteeInputs:string[] = ['name-input','school-input', 'major-input' ];
+    const mentorInputs:string[] =['profName-input','company-input','careerTitle-input', 'bio-input', 'specialty-input']
+
+    if (menteeForm && mentorForm && userTypeButton) {
+        if (value == 'Mentee') {
```

Comment:
maybe simplify this to:
```
const visibleId = value;
const hiddenId = visibleId == 'Mentee' ? 'Mentor' : 'Mentee';
showForm(visibleId);
hideForm(hiddenId)

```

Reply:
What exactly to simplify?

Reply:
Line 16?

Reply:
Can you replace the if/else blocks that are responsible for showing and hiding the forms with simpler function like the ones I suggested?

Reply:
yes

Reply:
Wouldn't it make logic more duplicated if the they break out into simpler functions just because we have to set attributes multiple times.

Reply:
Could you provide an example of the logic that would be duplicated?

Reply:
The logic would be duplicated from what was in the if else to the specific functions. 
Like inside show form it has: 
`

    if (visibleId == 'Mentee') {
        menteeForm.setAttribute('style', 'display:block;');
        userTypeButton.setAttribute('value', visibleId);
        setRequiredInputs(menteeInputs, true);
    }
    if (visibleId == 'Mentor') {
        mentorForm.setAttribute('style', 'display:block;');
        userTypeButton.setAttribute('value', visibleId);
        setRequiredInputs(mentorInputs, true);
    }
`

It's similar for hideForm, which creates morelines of code than there was before.

Reply:
Here's how the six lines can be written in three (this assumes required inputs is also passed into the method):

Line 1: document.getElementById(visibleId).setAttribute('style', 'display:block;')
Line 2: userTypeButton.setAttribute('value', visibleId);
Line 3: setRequiredInputs(requiredInputs, true);

Reply:
Oh, that's smart. I didn't think of that.

Reply:
Done